pkg/pillar/cmd/domainmgr/containerd.go

```diff
@@ -188,12 +272,210 @@ func getRktPodManifest(PodManifestFile string) (RktPodManifest, error) {
+	}
+	return ctrdClient, nil
+}
+
+func getContainerdContext(ctrdNameSpace string) context.Context {
+	ctrdCtx := namespaces.WithNamespace(context.Background(), ctrdNameSpace)
+	return ctrdCtx
+}
+
+// getContainerInfo - returns container config as JSON
+func getContainerInfo(ctrdCtx context.Context, container containerd.Container) (string, error) {
+	info, err := container.Info(ctrdCtx)
+	if err != nil {
+		return "", fmt.Errorf("Exception while fetching container info. %v", err)
+	}
+	if info.Spec != nil && info.Spec.Value != nil {
+		typeurl.Register(&specs.Spec{}, "types.containerd.io", "opencontainers/runtime-spec", "1", "Spec")
+		specValue, err := typeurl.UnmarshalAny(info.Spec)
+		if err != nil {
+			return "", fmt.Errorf("Exception while fetching container spec. %v", err)
+		}
+		if err != nil {
+			return "", fmt.Errorf("Exception while fetching container spec. %v", err)
+		}
+		res, err := getJSON(struct {
+			containers.Container
+			Spec interface{} `json:"Spec,omitempty"`
+		}{
+			Container: info,
+			Spec:      specValue,
+		})
+		if err != nil {
+			return "", fmt.Errorf("Exception while unmarshalling existing container spec. %v", err)
+		}
+		return res, nil
+	}
+	res, err := getJSON(info)
+	if err != nil {
+		return "", fmt.Errorf("Exception while unmarshalling newly added container spec. %v", err)
+	}
+	return res, nil
+}
+
+//createBundle - assigns a UUID and creates a bundle for container's rootFs
+func createBundle(ctrdCtx context.Context, ctrdClient *containerd.Client, container containerd.Container, snapshotName, imageConfigJSON string) error {
+	appDir := getContainerPath(container.ID())
+	rootFsDir := path.Join(appDir, containerRootfsPath)
+	//rootFsDir := getContainerRootfs(container.ID())
+	if err := os.MkdirAll(rootFsDir, 0766); err != nil {
+		return fmt.Errorf("Exception while creating rootFS dir. %v", err)
+	}
+
+	mountCommands, err := getBundleMountCommand(ctrdCtx, ctrdClient, snapshotName, rootFsDir)
+	if err != nil {
+		return fmt.Errorf("Exception while preparing snapshot mount. %v", err)
+	}
+	containerConfig, err := getContainerInfo(ctrdCtx, container)
+	if err != nil {
+		return fmt.Errorf("Exception while fetching container info. %v", err)
+	}
+	err = ioutil.WriteFile(filepath.Join(appDir, containerConfigFilename), []byte(containerConfig), 0666)
+	if err != nil {
+		return fmt.Errorf("Exception while writing container info to %v/%v. %v", appDir, containerConfigFilename, err)
+	}
+	err = ioutil.WriteFile(filepath.Join(appDir, imageConfigFilename), []byte(imageConfigJSON), 0666)
+	if err != nil {
+		return fmt.Errorf("Exception while writing image info to %v/%v. %v", appDir, imageConfigFilename, err)
+	}
+
+	err = ioutil.WriteFile(filepath.Join(appDir, pidFilename), []byte(container.ID()), 0666)
+	if err != nil {
+		return fmt.Errorf("Exception while writing container pid to %v/%v. %v", appDir, pidFilename, err)
+	}
+	_, err = executeShellCommands(mountCommands)
```

Comment:
Can we replace the exec-ing out to a shell with using direct library calls? [Mount](https://godoc.org/golang.org/x/sys/unix#Mount) should help